ext/telemetry/lib.rs

```diff
@@ -866,9 +969,596 @@ fn op_otel_span_flush(state: &mut OpState) {
+  };
+
+  match &mut temp.metric {
+    Some(TemporaryMetric {
+      data: TemporaryMetricData::Sum(sum),
+      ..
+    }) => sum.data_points.push(data_point),
+    Some(TemporaryMetric {
+      data: TemporaryMetricData::Gauge(gauge),
+      ..
+    }) => gauge.data_points.push(data_point),
+    _ => {}
+  }
+}
+
+#[op2(fast)]
+fn op_otel_metrics_histogram<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  state: &mut OpState,
+  name: v8::Local<'s, v8::Value>,
+  description: v8::Local<'s, v8::Value>,
+  unit: v8::Local<'s, v8::Value>,
+  #[smi] temporality: u8,
+) {
+  let Some(temp) = state.try_borrow_mut::<TemporaryMetricsExport>() else {
+    return;
+  };
+
+  if let Some(current_metric) = temp.metric.take() {
+    let metric = Metric::from(current_metric);
+    temp.scope_metrics.last_mut().unwrap().metrics.push(metric);
+  }
+
+  let name = owned_string(scope, name.cast());
+  let description = owned_string(scope, description.cast());
+  let unit = owned_string(scope, unit.cast());
+
+  let temporality = match temporality {
+    0 => Temporality::Delta,
+    1 => Temporality::Cumulative,
+    _ => return,
+  };
+  let histogram = opentelemetry_sdk::metrics::data::Histogram {
+    data_points: vec![],
+    temporality,
+  };
+
+  temp.metric = Some(TemporaryMetric {
+    name,
+    description,
+    unit,
+    data: TemporaryMetricData::Histogram(histogram),
+  });
+}
+
+#[allow(clippy::too_many_arguments)]
+#[op2(fast)]
+fn op_otel_metrics_histogram_data_point(
+  state: &mut OpState,
+  #[number] count: u64,
+  min: f64,
+  max: f64,
+  sum: f64,
+  start_time: f64,
+  time: f64,
+  #[smi] buckets: u32,
+) {
+  let Some(temp) = state.try_borrow_mut::<TemporaryMetricsExport>() else {
+    return;
+  };
+
+  let min = if min.is_nan() { None } else { Some(min) };
+  let max = if max.is_nan() { None } else { Some(max) };
+
+  let start_time = SystemTime::UNIX_EPOCH
+    .checked_add(std::time::Duration::from_secs_f64(start_time))
+    .unwrap();
+  let time = SystemTime::UNIX_EPOCH
+    .checked_add(std::time::Duration::from_secs_f64(time))
+    .unwrap();
+
+  let data_point = opentelemetry_sdk::metrics::data::HistogramDataPoint {
+    bounds: Vec::with_capacity(buckets as usize),
+    bucket_counts: Vec::with_capacity((buckets as usize) + 1),
+    count,
+    sum,
+    min,
+    max,
+    start_time,
+    time,
+    attributes: vec![],
+    exemplars: vec![],
+  };
+
+  if let Some(TemporaryMetric {
+    data: TemporaryMetricData::Histogram(histogram),
+    ..
+  }) = &mut temp.metric
+  {
+    histogram.data_points.push(data_point);
+  }
+}
+
+#[op2(fast)]
+fn op_otel_metrics_histogram_data_point_entry_final(
+  state: &mut OpState,
+  #[number] count1: u64,
+) {
+  let Some(temp) = state.try_borrow_mut::<TemporaryMetricsExport>() else {
+    return;
+  };
+
+  if let Some(TemporaryMetric {
+    data: TemporaryMetricData::Histogram(histogram),
+    ..
+  }) = &mut temp.metric
+  {
+    histogram
+      .data_points
+      .last_mut()
```

Comment:
Is it guaranteed there's gonna be at least one element?

Reply:
`op_otel_metrics_histogram_data_point` is always called before this